numpy/core/src/multiarray/methods.c

```diff
@@ -585,8 +585,9 @@ array_tostring(PyArrayObject *self, PyObject *args, PyObject *kwds)
 static PyObject *
 array_tofile(PyArrayObject *self, PyObject *args, PyObject *kwds)
 {
-    int own;
+    int own, res_write, res_close;
     PyObject *file;
+    PyObject *err_type = NULL, *err_value = NULL, *err_traceback = NULL;
```

Comment:
We use C99, so I'd prefer it if you put these declarations at their point of use:

```suggestion
    int own;
    PyObject *file;
```